includes/functions.php

```diff
@@ -474,3 +475,91 @@ function is_json( $data ) {
 function is_blog_public() {
 	return (bool) apply_filters( 'activitypub_is_blog_public', \get_option( 'blog_public', 1 ) );
 }
+
+/**
+ * Get active users based on a given duration
+ *
+ * @param string $duration The duration to check
+ *
+ * @return int The number of active users
+ */
+function get_active_users( $duration = '1 month ago' ) {
+	// get all distinct authors that have published a post in the last 30 days
+	$posts = get_posts(
+		array(
+			'post_type'      => 'post',
+			'post_status'    => 'publish',
+			'orderby'        => 'post_count',
+			'order'          => 'DESC',
+			'posts_per_page' => 4,
```

Comment:
Having this in the query will cause problems, I think? 

The query will be limited to 4 posts, which means that only a maximum of 4 users would be able to be active on a given site, and also if, say, one prolific author has published the 4 most recent posts, only that author will be counted?

Reply:
I think we can probably do this better with a custom SQL query. We would probably then also want to cache the query. I asked ChatGPT for help and it looks pretty promising:

```php
global $wpdb;
$query = "
  SELECT COUNT(DISTINCT post_author) 
  FROM {$wpdb->posts} 
  WHERE post_type = 'post' 
  AND post_status = 'publish'
  AND post_date >= DATE_SUB(NOW(), INTERVAL 1 MONTH)
";
$num_authors = $wpdb->get_var($query);
```

We could wrap it all in a transient with a `DAY_IN_SECONDS` expiry?

Reply:
Why have I added a posts per page? 😳🙄